### ISSUE TITLE
HAProxy Configuration Issues

### ISSUE THREAD

**Issue by AusIV** [NoteGio/openrelay#15]
OpenRelay uses HAProxy in front of the Ethereum RPC service, the intent being that if the state node or monitor node becomes unavailable, it can fall back to the standby node and eventually the RPC node serving the other purpose.

We're seeing a couple of issues with the current configuration. First, HAProxy is only resolving IPs at startup, so if a server goes down and comes back with a different address, HAProxy never routes requests there. Second, HAProxy won't come up until all of the Ethereum servers are available. So if the standby server is down and HAProxy gets restarted, HAProxy won't come back up until the standby node is back up, even though the primary node was available.

I believe this can be fixed in the HAProxy config, as discussed here: https://discourse.haproxy.org/t/haproxy-fails-to-start-if-backend-server-names-dont-resolve/322/17

Alternatively, if we can resolve https://github.com/NoteGio/openrelay/issues/8 I think we can just go to one big pool of Ethereum nodes and not need to use HAProxy at all. That's probably a larger topic though than tweaking the HAProxy config.

**Comment by AusIV** [comment]
No longer relevant after #22
